### PR TITLE
devise authentication_keys functionality

### PR DESCRIPTION
**Pull request by erickrause**
I updated routes and gemfile to be compatible with rails 3.1 and ruby 1.9.2
The integration tests were failing because it was comparing strings, which weren't the same, but had the same data.  So I converted them back to JSON objects for comparison.
Added functionality to map params passed in to devise authentication_keys.

### CODE REVIEW

File: lib/token_endpoint.rb [hunk 1]
```diff
@@ -33,7 +33,12 @@ class TokenEndpoint
       raise InvalidGrantType.new('invalid authorization code') unless code && code.valid_request?(req)
       client.refresh_tokens.create! :user => code.user
     when :password
-      resource = mapping.to.find_for_authentication(mapping.to.authentication_keys.first => req.username)
+      conditions = Hash.new
+      mapping.to.authentication_keys.each do |key|
+        conditions[key] = req.env["rack.request.form_hash"][key.to_s]
+      end
+      conditions[:email] = req.username
```
Comment (wireframe) [inline lib/token_endpoint.rb]: using the hardcoded conditions[:email] = req.username will break devise clients that do not use email as their primary lookup field (ex: username).

Comment (erickrause) [inline lib/token_endpoint.rb]: yep, good call.  changed it, and will submit a new pull request.

**Comment by erickrause** [comment]
Seems like it got added already.

**Comment by wireframe** [comment]
closing out as the oauth2 password spec is pretty explicit about username/password combo.  this may be a good option for an extension grant type.

**Pull request closed without merging**